graph/graph.go

```diff
@@ -153,6 +153,34 @@ func (self *Graph) pathHelper(dst godip.Province, queue []pathStep, seen map[[2]
 	return nil
 }
 
+func (self *Graph) reversePathHelper(src godip.Province, queue []pathStep, seen map[[2]godip.Province]bool, filter godip.PathFilter) []godip.Province {
+	var newQueue []pathStep
+	for _, step := range queue {
+		key := [2]godip.Province{step.src, step.dst}
+		if seen[key] {
+			continue
+		}
+		seen[key] = true
+		for name, edge := range self.reverseEdges(step.src) {
```

Comment:
Is this line the only difference between the two helpers?

If it is - just send the function to use when finding edges as a parameter to the helper instead of duplicating the entire function?